code/obj/machinery/drone_beacon.dm

```diff
@@ -0,0 +1,80 @@
+#define PLAYER_SEEK_RANGE 12
+#define DRONE_SEEK_RANGE 25
+
+/obj/machinery/drone_beacon
+	name = "Mysterious Beacon"
+	desc = "Some strange transmitter."
+	icon = 'icons/obj/ship.dmi'
+	icon_state = "beacon_synd"
+	anchored = TRUE
+	density = TRUE
+	processing_tier = PROCESSING_EIGHTH
+	_max_health = 500
+	_health = 500
+
+	New()
+		..()
+		src.AddComponent(/datum/component/obj_projectile_damage)
+		RegisterSignal(src, COMSIG_ATOM_HITBY_PROJ, .proc/hitby_proj)
+
+	ex_act(severity)
+		summon_drones()
+		..()
+
+	attackby(obj/item/I, mob/user)
+		summon_drones()
+		..()
+
+	disposing()
+		UnregisterSignal(src, COMSIG_ATOM_HITBY_PROJ)
+		..()
+
+	process()
+		var/list/target_list = list()
+		for(var/mob/M in range(PLAYER_SEEK_RANGE, src))
```

Comment:
It's not every process tick, it's currently at `PROCESSING_EIGTH`, but I can kick it down to `PROCESSING_SIXTEENTH` and slightly increase range if you think it will help?